### PR TITLE
List space

### PR DESCRIPTION
**Pull request by sabrinalord**
users can click on list a space, view the form to list space. Enter space details, name, price description. Info is sent to spaces database. Users preview listing on submit form click. Users can not input blank data into form, shows form error message.

### CODE REVIEW

File: spec/setup_test_database.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 def setup_test_database
   p "Setting up test database..."
 
-  DatabaseConnection.setup('makers_bnb_test')
-  DatabaseConnection.query('TRUNCATE users, spaces, bookings CASCADE;')
-  # DatabaseConnection.query("ALTER SEQUENCE users_id_seq RESTART WITH 1;")
+  connection = PG.connect(dbname: 'makers_bnb_test')
+  connection.exec('TRUNCATE users CASCADE;')
+  # Not how it's meant to be done - when merging, please keep original DatabaseConnection!
```
Comment (chris-clement) [inline spec/setup_test_database.rb]: Maybe one to discuss. What do you mean here pls?

File: spec/features/fill_in_listing_form_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require_relative 'listing_web_helper'
+
+feature 'filling in form' do
+
+  scenario 'user can fill in form' do
+    create_listing('My space')
```
Comment (chris-clement) [inline spec/features/fill_in_listing_form_spec.rb:6]: Is this a test? Not seeing an expect statement so not sure if this is actually doing something

File: lib/makersbnb.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+require 'pg'
+
+class MakersBnb_Listings
+
+  def self.create_space(space_name:, price:, description:)
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'makers_bnb_test')
```
Comment (chris-clement) [inline lib/makersbnb.rb:7]: we have a setup_database.rb and lib/database_connection.rb which would help to clean up this code. Nothing to change for now but once merged we could make use of these files.

File: lib/makersbnb.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def self.exist?(space_name:)
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'makers_bnb_test')
+    else
+      connection = PG.connect(dbname: 'makers_bnb')
+    end
+    result = connection.exec_params("SELECT * FROM spaces WHERE name = $1;",[space_name])
+    @unsuccessful = !result.to_a.empty?
```
Comment (chris-clement) [inline lib/makersbnb.rb:22]: Can you please explain what is being done here? I think we have something similar and used result.count.zero? Which returns true if the sql query being run has no results.

**Review by chris-clement — COMMENTED** [review]
Looking good! Few comments from me

File: spec/features/fill_in_listing_form_spec.rb (hunk 2 above)
Comment (Hereloss) [reply spec/features/fill_in_listing_form_spec.rb:6]: This is just testing the form works - kind of redundant with the other tests now though, so may be worth removing

File: spec/setup_test_database.rb (hunk 1 above)
Comment (Hereloss) [reply spec/setup_test_database.rb]: We just didn't make the Database Connection file which you guys have, but we can edit this once merged!

File: lib/makersbnb.rb (hunk 4 above)
Comment (Hereloss) [reply lib/makersbnb.rb:22]: Yeah pretty similar I think, we're just checking if the space name has already been taken or if it's free

File: spec/setup_test_database.rb (hunk 1 above)
Comment (sabrinalord) [reply spec/setup_test_database.rb]: this was just a note to say we need to create our DatabaseConnection class to neaten this up as you did in yours

File: lib/makersbnb.rb (hunk 4 above)
Comment (sabrinalord) [reply lib/makersbnb.rb:22]: yes agree result.count.zero? would improve readability

File: spec/setup_test_database.rb (hunk 1 above)
Comment (chris-clement) [reply spec/setup_test_database.rb]: Ahh great, thanks for clarifying!

**Pull request merged**